File: app/src/main/java/br/com/jonathanzanella/myexpenses/card/CardContract.java
package br.com.jonathanzanella.myexpenses.card;

import android.content.Context;
import android.support.annotation.StringRes;

import br.com.jonathanzanella.myexpenses.account.Account;
import br.com.jonathanzanella.myexpenses.validations.ValidationError;

/**
 * Created by jzanella on 8/28/16.
 */

interface CardContract {
	interface View {
		Context getContext();
		void setTitle(@StringRes int string);
		void setTitle(String string);
		void showCard(Card card);
	}

	interface EditView extends View {
		Card fillCard(Card card);
		void finishView();
		void showError(ValidationError error);
		void onAccountSelected(Account account);
	}
}

File: app/src/main/java/br/com/jonathanzanella/myexpenses/card/CardContract.kt
package br.com.jonathanzanella.myexpenses.card

import android.content.Context
import android.support.annotation.StringRes

import br.com.jonathanzanella.myexpenses.account.Account
import br.com.jonathanzanella.myexpenses.validations.ValidationError

internal interface CardContract {
    interface View {
        val context: Context
        fun setTitle(@StringRes string: Int)
        fun setTitle(string: String)
        fun showCard(card: Card)
    }

    interface EditView : View {
        fun fillCard(card: Card): Card
        fun finishView()
        fun showError(error: ValidationError)
        fun onAccountSelected(account: Account)
    }
}
